src/presentation/api/authentication/AuthenticationController.test.ts

```diff
@@ -61,33 +62,87 @@ describe('auth endpoints', () => {
     });
   });
 
-  describe('POST /auth/login', () => {
-    it('returns 422 if the code does not exist', async () => {
-      await request(app)
-        .post('/api/v1/auth/login')
-        .send({ method: 'magic-link', email: 'some@email.com', authCode: v4() })
-        .expect(422)
-        .expect((res) => {
-          expect(res.body.code).toEqual(AuthorisationFailureReason.AUTH_CODE_NOT_FOUND);
-        });
-    });
-
-    it('exchanges an auth code for a signed JWT', async () => {
-      const magicLink = await createMagicLink.execute('kostas@example.com');
+  describe('POST /auth/sessions', () => {
+    it('creates a Dokobit session', async () => {
+      nock('https://id-sandbox.dokobit.com')
+        .post('/api/authentication/create', { return_url: /\/authenticate/i })
+        .query({ access_token: 'dummy' })
+        .reply(200, { session_token: 'some session token', url: 'example.com/dokobit-redirect' });
 
       await request(app)
-        .post('/api/v1/auth/login')
-        .send({
-          method: 'magic-link',
-          email: magicLink.email.value,
-          authCode: magicLink.code.value,
-        })
+        .post('/api/v1/auth/sessions')
         .expect(200)
         .expect((response) => {
-          expect(response.body.token).toBeDefined();
+          expect(response.body.redirectUrl).toBe('example.com/dokobit-redirect');
         });
     });
   });
+
+  describe('POST /auth/login', () => {
+    describe('with magic link', () => {
+      it('returns 422 if the code does not exist', async () => {
+        await request(app)
+          .post('/api/v1/auth/login')
+          .send({ method: 'MAGIC_LINK', authCode: v4() })
+          .expect(422)
+          .expect((res) => {
+            expect(res.body.code).toEqual(MagicLinkAuthenticationErrorReason.AUTH_CODE_NOT_FOUND);
+          });
+      });
+
+      it('exchanges an auth code for a signed JWT', async () => {
+        const magicLink = await createMagicLink.execute('kostas@example.com');
+
+        await request(app)
+          .post('/api/v1/auth/login')
+          .send({
+            method: 'MAGIC_LINK',
+            authCode: magicLink.code.value,
+          })
+          .expect(200)
+          .expect((response) => {
+            expect(response.body.token).toBeDefined();
+          });
+      });
```

Comment:
Could we add a test here to see that it fails if we're in Estonian ID mode?

Reply:
Perhaps similar for the opposite scenario as well, shouldn't be possible to start an Estonian ID session if we're on a magic link deployment.

Reply:
We have not yet added any kind of enforcement. I think we should as this will clean up the authentication factory stuff as well